packages/core/js-sdk/src/client.ts

```diff
@@ -125,59 +152,92 @@ export class Client {
       )
 
       // Any non-request errors (eg. invalid JSON in the response) will be thrown as-is.
-      return fetch(normalizedInput, normalizeRequest(init, headers)).then(
-        (resp) => {
-          this.logger.debug(`Received response with status ${resp.status}\n`)
-          return normalizeResponse(resp, headers)
-        }
-      )
+      return fetch(
```

Comment:
todo: add await, just to prevent miss behaviour inside other functions and try catch block just in case